### PR TITLE
use absolute exclude path in `build/webpack.test.config.js` and fix error when recursively traverse an object

### PR DESCRIPTION
**Pull request by flytreeleft**
* use absolute path to make sure running tests is OK even if it is in node_modules of other project
* fix 'Maximum call stack size exceeded' when recursively traverse an object

### CODE REVIEW

File: src/watcher.js [hunk 1]
```diff
@@ -331,14 +331,25 @@ Watcher.prototype.teardown = function () {
  * @param {*} val
  */
 
-function traverse (val) {
+function traverse (val, walkedObjs) {
   var i, keys
+
+  !walkedObjs && (walkedObjs = {})
```
Comment (posva) [inline src/watcher.js]: I'm curious about this notation. How is it different from `walkedObjs = walkedObjs || {}` ?

Comment (flytreeleft) [inline src/watcher.js]: Uh, it's just personal habits, there is no difference between them, do not be feeling trouble with it.

Comment (posva) [inline src/watcher.js]: I wasn't complaining, just curious :smile:

**Comment by yyx990803** [comment]
Please do not mix multiple fixes in the same PR, since this makes it harder to review and merge. Submit separate PRs please.

**Comment by flytreeleft** [comment]
Splitting to #2686 and #2685

**Pull request closed without merging**